webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/InterpolatedValueCompiler.scala

```diff
@@ -406,64 +405,88 @@ class InterpolatedValueCompilerImpl extends RegexParsers with InterpolatedValueC
+                                                propOption.? ).map { case (path, opt) => Property(path.toList, opt) }
+
+  // other cases of ${}: cfengine variables, etc
+  def otherProp[_: P]: P[NonRudderVar] = P( CharsWhile(c => c != '}').!).map { s => NonRudderVar("${"+s+"}") }
 
   // all interpolation
-  val interpol    : Parser[Interpolation] = rudderProp | nodeProp
+  def interpol [_: P]   : P[Interpolation] = P( rudderProp | nodeProp )
 
   //here, the number of " must be strictly decreasing - ie. triple quote before
-  val propOption  : Parser[PropertyOption] = space ~> "|" ~> space ~>  ( onNodeOpt |  "default" ~> space ~> "=" ~> space ~>
-                                             ( interpolOpt | emptyTQuote | tqString | emptySQuote | sqString )  )
+  def propOption[_: P]  : P[PropertyOption] = P( s ~ "|" ~/ s ~ ( onNodeOpt | defaultOpt ) )
 
-  val onNodeOpt   : Parser[InterpreteOnNode.type] = id("node") ^^^ InterpreteOnNode
+  def defaultOpt[_: P]  : P[DefaultValue] = P( IgnoreCase("default") ~/ s ~ "=" ~/ s ~/ ( interpolOpt | emptyTQuote | tqString | emptySQuote | sqString ) )
+  def onNodeOpt[_: P]   : P[InterpreteOnNode.type] = P( IgnoreCase("node")).map(_ => InterpreteOnNode)
 
-  val interpolOpt : Parser[DefaultValue] = interpol ^^ { x => DefaultValue(x::Nil) }
-  val emptySQuote : Parser[DefaultValue] = id("\"\"")         ^^   { _ => DefaultValue(CharSeq("")::Nil) }
-  val emptyTQuote : Parser[DefaultValue] = id("\"\"\"\"\"\"") ^^   { _ => DefaultValue(CharSeq("")::Nil) }
+  def variable[_: P] = P("${" ~ s ~/ varname ~/ s ~ "}")
+  def interpolOpt[_: P] : P[DefaultValue] = variable.map { x => DefaultValue(x::Nil) }
+  def emptySQuote[_: P] : P[DefaultValue] = P( "\"\"" ).map { _ => DefaultValue(CharSeq("")::Nil) }
+  def emptyTQuote[_: P] : P[DefaultValue] = P("\"\"\"\"\"\"").map { _ => DefaultValue(CharSeq("")::Nil) }
 
-  //string must be simple or triple quoted string
-  val sqString    : Parser[DefaultValue] = id("\"")  ~> rep1(sqplainStr | interpol )<~ id("\"") ^^ { case x => DefaultValue(x) }
-  val sqplainStr  : Parser[Token]        = ("""(?iums)((?!(\Q${\E\s*rudder\s*\.|\Q${\E\s*node\s*\.\s*properties|")).)+""").r  ^^ { str => CharSeq(str) }
 
-  val tqString    : Parser[DefaultValue]   = id("\"\"\"")  ~> rep1(tqplainStr | interpol )<~ id("\"\"\"") ^^ { case x => DefaultValue(x) }
-  val tqplainStr  : Parser[Token]          = ("""(?iums)((?!(\Q${\E\s*rudder\s*\.|\Q${\E\s*node\s*\.\s*properties|"""+"\"\"\""+ """)).)+""").r  ^^ { str => CharSeq(str) }
+  def leftTag[_: P] = P( "<" ~ (!">" ~ AnyChar).rep(1).! ~ ">" )
+
+  //string must be simple or triple quoted string
+  def sqString[_: P]    : P[DefaultValue] = P( "\"" ~ (sqplainStr | variable ).rep(1) ~ "\"").map { case x => DefaultValue(x.toList) }
+  def sqplainStr[_: P]  : P[Token]        = P( CharsWhile(c => c != '"').! ).map { str => CharSeq(str) }
```

Comment:
This does not managed escaped quote

Reply:
also, i thin it will ignores variables, it should be something like the begining 

(plainStringNoDollar | varOrString)

But cannot use them, because the condition of underlying strings are not the same (no quote character)

Reply:
we never managed escaped quote in node properties default, we have triple quote for that. If you think of a use case node supported, I should go in an other bug report I believe

Reply:
ok no need to support escaped quote, if it was not like before 

But you are ignoring variables in string 

default=" defaut ${node.properties[datacenterDefault]}" should match Default(CharSeq(" defaut "), Property(datacenterDefault)) but it is a CharSeq" defaut ${node.properties[datacenterDefault]}"

I added a test case in my PR

Reply:
That pr fixes mosr problems with inner variables in quotes, i also need to look some other edge cases (non ending variables, non Rudder variable too https://github.com/fanf/rudder/pull/10/files